GetMigrationsStatusByDelegate/handler.ts

```diff
@@ -0,0 +1,147 @@
+import { MigrationsByDelegate } from "../utils/query";
+import { MigrationsStatusByDelegateResponse } from "../generated/definitions/MigrationsStatusByDelegateResponse";
+
+// set Postgres as default db target for the query builder
+const knex = knexBase({
+  client: "pg"
+});
+
+type Handler = (
+  context: Context,
+  delegateId: NonEmptyString
+) => Promise<
+  | IResponseSuccessJson<MigrationsStatusByDelegateResponse>
+  | IResponseErrorInternal
+  | IResponseErrorNotFound
+>;
+
+/*
+We get a resultSet with this fields for every Organization the Delagate owns/owned a subscrition
+organizationFiscalCode | Initial | Processing | Failed | Completed
+
+where the latest 4 fields are the sum for every subscriptions in the specific status
+*/
+export const createSqlStatus = (dbConfig: IDecodableConfigPostgreSQL) => (
+  delegateId: string
+): NonEmptyString =>
+  MigrationsByDelegate(dbConfig, delegateId)
+    .select([
+      "organizationFiscalCode",
+      knex.raw(`max("updateAt") as "lastUpdate"`),
+      knex.raw(
+        `sum(CASE WHEN "status" = 'INITIAL' THEN 1 ELSE 0 END) as initial`
+      ),
+      knex.raw(
+        `sum(CASE WHEN "status" = 'PROCESSING' THEN 1 ELSE 0 END) as processing`
+      ),
+      knex.raw(
+        `sum(CASE WHEN "status" = 'FAILED' THEN 1 ELSE 0 END) as failed`
+      ),
+      knex.raw(
+        `sum(CASE WHEN "status" = 'COMPLETED' THEN 1 ELSE 0 END) as completed`
+      )
+    ])
+    .groupBy(["organizationFiscalCode"])
+    .orderBy("lastUpdate", "desc")
+    .toQuery() as NonEmptyString;
+
+/*
+We would retrieve all the operations for each organization for which the delegate owns/owned at least one subscription
+ */
+export const getMigrationByDelegate = (config: IConfig, connect: Pool) => (
+  delegateId: string
+): TE.TaskEither<IDbError, QueryResult> =>
+  pipe(
+    createSqlStatus(config)(delegateId),
+    sql => queryDataTable(connect, sql),
+    TE.mapLeft(flow(toPostgreSQLErrorMessage, toPostgreSQLError))
+  );
+
+// format a db recordset into a struct as expected by the response specification
+export const processResponseFromResultSet = (
+  resultSet: QueryResult
+): E.Either<t.Errors, MigrationsStatusByDelegateResponse> =>
+  pipe(
+    resultSet.rows,
+    RA.map(row => ({
+      lastUpdate: row.lastUpdate,
+      organization: {
+        fiscalCode: row.organizationFiscalCode
+      },
+      status: {
+        completed: Number(row.completed),
+        failed: Number(row.failed),
+        initial: Number(row.initial),
+        processing: Number(row.processing)
+      }
+    })),
+    items => ({ items }),
+    MigrationsStatusByDelegateResponse.decode
+  );
+
+export const createHandler = (config: IConfig, pool: Pool): Handler => async (
+  _context,
+  delegateId
+): ReturnType<Handler> =>
+  pipe(
+    getMigrationByDelegate(config, pool)(delegateId),
+    TE.mapLeft(e =>
+      ResponseErrorInternal(`Failed to execute query on database: ${e.message}`)
+    ),
+    TE.chainW(
```

Comment:
Maybe we can use `chain` instead of `chainW`

Reply:
good catch! thanks